code/part-two/client/source/MojiItem.jsx

```diff
@@ -1,13 +1,33 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
 
-export function MojiItem(props) {
-  console.log('RENDERING: <MojiItem />');
-  return (
-    <div>
-      <Link to={'/moji/' + props.address}>
-        {props.address}
-      </Link>
-    </div>
-  );
+import { getMoji } from './services/requests';
+import { parseDna } from './services/parseDna';
+
+export class MojiItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      mojiView: null
+    };
+  }
+
+  componentDidMount() {
+    getMoji(this.props.address)
+      .then(moji => {
+        this.setState({
+          mojiView: parseDna(moji.dna).view
+        });
+      });
+  }
+
+  render() {
+    return (
+      <div>
+        <Link to={'/moji/' + this.props.address}>
```

Comment:
I had used `this.props.history.push` for redirects in my previous commits. This looks like a better option. I will make sure to change it in the signup component for consistencies sake.

Reply:
I _think_ using `this.props.history.push` is good for buttons! But I used `<Link>` here as it's pretty much the `<a>` equivalent.

Reply:
Oh right these are anchors, still brushing up on the react stuff.